kale/pipeline/fewshot_trainer.py

```diff
@@ -0,0 +1,124 @@
+# =============================================================================
+# Author: Wenrui Fan, winslow.fan@outlook.com
+# =============================================================================
+
+"""ProtoNet trainer (pipelines)
+
+This module contains the ProtoNet trainer class and its related functions. It trains the ProtoNet model in N-way-k-shot problems.
+
+This module uses `PyTorch Lightning <https://github.com/Lightning-AI/lightning>` to standardize the workflow.
+
+This is a modified version of the original prototypical neural networks for few-shot learning projects from https://github.com/jakesnell/prototypical-networks.
+"""
+
+from typing import Any
+
+import pytorch_lightning as pl
+import torch
+
+from kale.predict.losses import protonet_loss
+
+
+class ProtoNetTrainer(pl.LightningModule):
+    """ProtoNet trainer class
+
+    Args:
+        model (torch.nn.Module): A feature extractor replaced classfier with a flatten layer. Output 1-D feature vectors.
```

Comment:
The grammar is a bit problematic. Make it clearer please.

Reply:
Change to _A feature extractor without any task-specific heads_.